core/src/main/java/org/infinispan/reactive/publisher/impl/PublisherRequestCommand.java

```diff
@@ -145,10 +145,8 @@ public boolean isReturnValueExpected() {
 
    @Override
    public boolean canBlock() {
-      // When the stream isn't parallel the data container iteration is done by the calling thread - and the user
-      // provides the transformation, so it could block - this technically slows down sequential for some cases, but
-      // is much safer
-      return !parallelStream;
+      // This command is guaranteed to only use CPU now - stores are done in a blocking thread pool
+      return false;
```

Comment:
AFAICT the calling thread can still invoke user code when it's stealing a segment, even when `parallelStream=true`

Reply:
They can't yet, as the API is not exposed :) This is an issue with ensuring that the user do non blocking code only in their operations, which would have to be documented in the forthcoming API. /cc @karesti 

I don't even know if the Publisher will be directly exposed yet or it will only be Ickle string.

Reply:
I was reading the old comment, are you saying it was never true?

       // When the stream isn't parallel the data container iteration is done by the calling thread - and the user
      // provides the transformation, so it could block - this technically slows down sequential for some cases, but

Reply:
Technically it is true, but it looks like as things progress that we may not even expose the publisher. And even if we do I mentioned we should state that the method should not block, just like we do with listeners.

Reply:
Huh, I assumed the new API that Katia is working on will be all about publishers and won't expose `entrySet` any more :)

I agree that the subscribers should never block, and that should be documented, I was just surprised that the old comment identified user code as the potential source of blocking and the new comment doesn't mention user code at all.

Reply:
That was written a long time ago, back before I even completed non blocking listeners. In that PR specifically you convinced me that users should be allowed to be smart enough to ensure their code is not blocking. I am just taking the same stance here. My previous stance was way more skeptical.

Reply:
It also still doesn't take into account that quite possibly users will never be able to supply these arguments themselves. If that becomes an issue I would say we can revisit that later if needed.